relay-server/src/actors/store.rs

```diff
@@ -254,6 +288,52 @@ impl StoreService {
         Ok(())
     }
 
+    fn handle_store_metrics(&self, message: StoreMetrics) {
+        let StoreMetrics {
+            buckets,
+            scoping,
+            retention,
+            mode,
+        } = message;
+
+        let batch_size = self.config.metrics_max_batch_size_bytes();
+        let mut dropped = SourceQuantities::default();
+        let mut error = None;
+
+        for bucket in buckets {
+            // Create a local bucket view to avoid splitting buckets unnecessarily. Since we produce
+            // each bucket separately, we only need to split buckets that exceed the size, but not
+            // batches.
+            for view in BucketsView::new(&[bucket]).by_size(batch_size).flatten() {
```

Comment:
nit: Could hide this behind `bucket.split(max_size)`, which returns the iterator.

Reply:
Let's leave this for a follow-up, I believe we might need such a signature anyway for the cross-org endpoint.